Makefile

```diff
@@ -175,47 +144,29 @@ shell: build-dirs build-env
 		$(BUILDER_IMAGE) \
 		/bin/sh $(CMD)
 
-DOTFILE_IMAGE = $(subst :,_,$(subst /,_,$(IMAGE))-$(VERSION))
-
-all-containers:
-	$(MAKE) container
-	$(MAKE) container BIN=velero-restic-restore-helper
-
-container: local-arch .container-$(DOTFILE_IMAGE) container-name
-.container-$(DOTFILE_IMAGE): _output/bin/$(GOOS)/$(GOARCH)/$(BIN) $(DOCKERFILE)
-	@cp $(DOCKERFILE) _output/.dockerfile-$(BIN)-$(GOOS)-$(GOARCH)
-	@docker build --pull -t $(IMAGE):$(VERSION) -f _output/.dockerfile-$(BIN)-$(GOOS)-$(GOARCH) _output
-	@docker images -q $(IMAGE):$(VERSION) > $@
-
-container-name:
+container-build-env:
+	@docker buildx build \
+	--target=builder-env \
+	--build-arg=PKG=$(PKG) \
+	--build-arg=VERSION=$(VERSION) \
+	--build-arg=GIT_SHA=$(GIT_SHA) \
+	--build-arg=GIT_TREE_STATE=$(GIT_TREE_STATE) \
+	-f Dockerfile .
+
+container:
```

Comment:
The default buildx builder uses the `docker` driver, which does indeed not support auto-push. The Workflow environment uses the `docker-container` driver. You can configure your local environment to use `docker-container` as well by doing this:
`docker buildx create --name builder --driver docker-container --use` and then:
`docker buildx inspect --bootstrap` to verify the builder is configured properly.
See also https://medium.com/@artur.klauser/building-multi-architecture-docker-images-with-buildx-27d80f7e2408 for more information. I am using the docker image based installation, and not the host based installation.

I am running on Linux, so I can't say if there are any special steps to take on Darwin.

Reply:
Alternatively, you can just remove the `--output=type=registry` from the `Makefile` on line 158. All the layers will still be built, but not pushed. However, due to the way buildx works, you will also not have an image as the docker daemon currently has no support for loading a multi-arch image, so `--load` doesn't work. What you might be able to do is output to "local" so the image is written as a file. See: https://github.com/docker/buildx#-o---outputpath-typetypekeyvalue

Reply:
OK so the instructions seem to be the same for darwin as well.
```
docker buildx create ...
```
We'll want these instructions documented. https://velero.io/docs/v1.4/development/ would be a good place for it.

After that is done, I am seeing this error
```
 => ERROR [linux/ppc64le stage-2 2/3] RUN apt-get update && apt-get install -y ca-certificates && rm -rf /var/lib/apt/lists/*                         1.6s
 => CANCELED [linux/arm/v7 stage-2 2/3] RUN apt-get update && apt-get install -y ca-certificates && rm -rf /var/lib/apt/lists/*                       0.6s
 => CANCELED [linux/amd64 stage-2 2/3] RUN apt-get update && apt-get install -y ca-certificates && rm -rf /var/lib/apt/lists/*                        0.4s
------
 > [linux/ppc64le stage-2 2/3] RUN apt-get update && apt-get install -y ca-certificates && rm -rf /var/lib/apt/lists/*:
#20 1.242 qemu: uncaught target signal 4 (Illegal instruction) - core dumped
#20 1.251 Illegal instruction
------
failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c apt-get update && apt-get install -y ca-certificates && rm -rf /var/lib/apt/lists/*]: buildkit-runc did not terminate successfully
```
Any ideas on what this might me?

Reply:
@ashish-amarnath sounds like OSX does not play nicely with the ppc64le image. What is the output of `docker buildx inspect --bootstrap`?

Reply:
Debugged this offline w/ @RobReus.
Building for `linux/ppc64le` doesn't work on OSX. We might have to exclude `linux/ppc64le` from building on OSX but build that in the CI.